ducktape/services/service.py

```diff
@@ -69,23 +68,58 @@ def __repr__(self):
                              (self.num_nodes, [n.account.hostname for n in self.nodes]))
 
     @property
+    def name(self):
+        """Human-readable identifier (almost certainly) unique within a test run."""
+        return self.__class__.__name__ + "-" + str(self.id)
```

Comment:
Maybe `name` is the wrong word for this property, and maybe we don't need both an "id" and a "name", but the idea is it's useful to have unique but human-readable identifier of a service instance

Use cases I had in mind:
* folder into which log files are collected after running a test
  * There is a currently a bug where if a test has multiple instances of a service, the logs get collected underneath the same folder.
  * E.g. mirror maker system tests in kafka
* It is useful having a unique service name when creating a path on remote machines where logs etc should be stored